workspaces/orchestrator/plugins/orchestrator-backend/src/service/SonataFlowService.ts

```diff
@@ -105,11 +105,24 @@ export class SonataFlowService {
     const urlToFetch = args.businessKey
       ? `${args.serviceUrl}/${args.definitionId}?businessKey=${args.businessKey}`
       : `${args.serviceUrl}/${args.definitionId}`;
+    const headers: Record<string, string> = {
+      'Content-Type': 'application/json',
+    };
+
+  // Add X-Authentication headers from authTokens
+  if (args.inputData && Array.isArray(args.inputData.authTokens)) {
+    args.inputData.authTokens.forEach((tokenObj: Record<string, unknown>) => {
+      if (tokenObj.provider && tokenObj.token) {
+        const headerKey = `X-Authentication-${tokenObj.provider}`;
+        headers[headerKey] = String(tokenObj.token); // Ensure token is a string
+      }
+    });
+  }
```

Comment:
it may be nice to have an `else` statement with a log message stating there was no authTokens provided, it may come in handy in case of issue